fees/beraborrow/index.ts

```diff
@@ -0,0 +1,233 @@
+      ? logs.map((log: any) => log.collateralFee)
+      : [0];
+  })
+);
+
+dailyFees.add(assets, DenManagerRedemptionFees.flat());
+
+
+
+
+// Liquidation fee (Debt gas compensation - TEMPORARY)
+
+const logs = await options.getLogs({
+  target: NECT,
+  eventAbi: "event Transfer(address indexed from, address indexed to, uint256 value)",
+  topics: [
+    TRANSFER_TOPIC,
+    "0x000000000000000000000000" + NectGasPool.substring(2),
+    "0x000000000000000000000000" + FEE_RECEIVER.substring(2),
+  ],
+  fromBlock: 10
+})
+
+const total = logs.reduce((acc: BigNumber, log: any) => {
+    return acc.plus(new BigNumber(log[2]));
+  }, new BigNumber(0));
+
+dailyFees.add(NECT, total);
+
+// Liquidation fee (Collateral gas compensation - TEMPORARY)
+
+await Promise.all(
+    
+  tuples.map(async (tuple) => {
+    const denManagers = tuple.denManagers;
+    const collateral = tuple.collateral;
+
+    const logs = await options.getLogs({
+      target: collateral,
+      eventAbi: "event Transfer(address indexed from, address indexed to, uint256 value)",
+      topics: [
+        TRANSFER_TOPIC,
+        "0x000000000000000000000000" + denManagers[0].substring(2),
+        "0x000000000000000000000000" + FEE_RECEIVER.substring(2),
+      ],
+    });
+    
+    const total = logs.reduce((acc: BigNumber, log: any) => {
+      return acc.plus(new BigNumber(log[2]));
+    }, new BigNumber(0));
+
+    dailyFees.add(collateral, total);
+  })
+);
+
+  
+// PermissionlessPSM deposit fee (in NECT)
+
+await Promise.all(
+  PSMContracts.map(async (psm) => {
+    const logs = await options.getLogs({
+      target: psm,
+      eventAbi: "event Deposit(address indexed caller, address indexed stable, uint stableAmount, uint mintedNect, uint fee)",
+    });
+
+    const total = logs.reduce((acc, log) => acc.plus(new BigNumber(log[4])), new BigNumber(0));
+    if (total.isGreaterThan(0)) dailyFees.add(NECT, total);
+  })
+);
+
+
+
+// PermissionlessPSM withdrawal fee (fee is in indexed stable)
+// OLD AND new PSM
+
+await Promise.all(
+  PSMContracts.map(async (psm) => {
+    const logs = await options.getLogs({
+      target: psm,
+      eventAbi: "event Withdraw(address indexed caller, address indexed stable, uint stableAmount, uint burnedNect, uint fee)",
+      fromBlock: 1470110,
+    });
+
+    logs.forEach((log) => {
+      const stable = log[1];
+      const fee = log[4];
+      if (stable && fee) dailyFees.add(stable, fee);
+    });
+  })
+);
+
+
+// LSP deposit and withdrawal fee
+
+  const tokens = await options.api.call({
+    target: LSPRouter,
+    abi: "function lspUnderlyingTokens() view returns (address[])",
+  });
+
+  tokens.map(async (token) => {
+    const logs = await options.getLogs({
+      target: token,
+      eventAbi: "event Transfer(address indexed from, address indexed to, uint256 value)",
+      topics: [
+        TRANSFER_TOPIC,
+        zeroAddress,
+        "0x000000000000000000000000" + FEE_RECEIVER.substring(2),
```

Comment:
again, room for doublecounting if some other event also sends to the fee receiver?